src/main/java/alchemyplusplus/gui/GUIHandler.java

```diff
@@ -14,34 +14,31 @@
 import cpw.mods.fml.common.network.IGuiHandler;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.world.World;
-
 import java.util.HashMap;
+import net.minecraft.tileentity.TileEntity;
 
 public class GUIHandler implements IGuiHandler
 {
-
     public static HashMap<String, Integer> guiHandlerRegistry = new HashMap<String, Integer>();
 
     @Override
     public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z)
     {
-        System.out.println("Got to the client GUI element method");
+        TileEntity entity = world.getTileEntity(x, y, z);
+
         switch (ID)
         {
             case 0:
             { // Potion mixer
-                FluidMixerTileEntity entity = (FluidMixerTileEntity) world.getTileEntity(x, y, z);
-                return new FluidMixerGUI(player.inventory, entity);
+                return new FluidMixerGUI(player.inventory, (FluidMixerTileEntity) entity);
```

Comment:
Is there any reason to declare the Block as a Tile Entity _BEFORE_ the player interacts with it instead of _WHEN_? Because if I understand the code edits, that's what happens, though I'm unsure about the side effects of such actions.